iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -90,12 +100,11 @@ public void AddOrUpdate(IDictionary<string, object> properties, string component
         /// Adds or updates the value for the collection.
         /// </summary>
         /// <remarks>
-        /// When adding a type of <see cref="IWritablePropertyResponse"/> the <see cref="ObjectSerializer.CheckWritablePropertyResponseType(object)"/>
-        /// checks to make sure it can serialize this type correctly.
-        /// This will only be evaluated at runtime so it will throw an exception if the type does not pass the check.
+        /// When using this as part of the <see cref="DeviceClient.SubscribeToWritablePropertyEventAsync(Func{ClientPropertyCollection, object, System.Threading.Tasks.Task}, object, System.Threading.CancellationToken)"/> flow. You should use the <see cref="PayloadConvention.CreateWritablePropertyResponse(object, int, long, string)"/> method to add the correct <see cref="IWritablePropertyResponse"/> to ensure the correct formatting is applied when the object is serialized.
```

Comment:
how about instead of mentioning the subscription flow, we instead mention that the response needs to be of the correct format when responding to a writable property update request?

`When using this as part of the <see cref="DeviceClient.UpdatePropertiesAsync(ClientPropertyCollection, System.Threading.CancellationToken)"/> flow to respond to a writable property update request, you should use the <see cref="PayloadConvention.CreateWritablePropertyResponse(object, int, long, string)"/> method to add the correct <see cref="IWritablePropertyResponse"/> to ensure the correct formatting is applied when the object is serialized.`